src/components/YourFeed.tsx

```diff
@@ -151,6 +154,53 @@ export class YourFeed extends React.PureComponent<DispatchProps & StateProps, Yo
         );
     }
 
+    private NavHeader(props: DispatchProps & StateProps &
+        { onFollowPressed: (author: Author) => void, isFavorite: () => boolean }) {
+        switch (props.yourFeedVariant) {
+            case 'favorite': {
+                return (
+                    <NavigationHeader
+                        leftButtonText=' '
+                        title={props.navigation.state.params ? props.navigation.state.params.author.name : 'Favorites'}
+                    />
+                );
+            }
+            case 'feed': {
+                const isFollowedFeed = props.navigation.state.params != null &&
+                    props.feeds.find(feed => feed.feedUrl === props.navigation.state.params.author.uri) != null;
+                return (
+                    <NavigationHeader
+                        onPressLeftButton={() => props.navigation.goBack(null)}
+                        rightButtonText1={props.notOwnFeed ? <Icon
+                            name={isFollowedFeed ? 'link-variant-off' : 'link-variant'}
+                            size={20}
+                            color={Colors.DARK_GRAY}
+                        /> : undefined}
+                        rightButtonText2={props.notOwnFeed ? <MaterialIcon
+                            name={'favorite'}
+                            size={20}
+                            color={isFollowedFeed
+                                ? props.isFavorite() ? Colors.BRAND_RED : Colors.DARK_GRAY
+                                : 'transparent'
+                            }
+                        /> : undefined}
+                        onPressRightButton1={async () => props.notOwnFeed && await props.onFollowPressed(props.navigation.state.params.author)}
+                        onPressRightButton2={() => props.notOwnFeed && isFollowedFeed && props.onToggleFavorite(props.navigation.state.params.author.uri)}
+                        title={props.navigation.state.params ? props.navigation.state.params.author.name : 'Favorites'}
```

Comment:
It would be better to use a variable for `author` and not use the whole `props.navigation.state.params.author...` everywhere

Reply:
👍